src/dtx/dtx_common.c

```diff
@@ -236,10 +237,12 @@ dtx_batched_commit(void *arg)
 		ABT_thread_yield();
 	}
 
-	while (!d_list_empty(&dmi->dmi_dtx_batched_list)) {
-		dbca = d_list_entry(dmi->dmi_dtx_batched_list.next,
-				    struct dtx_batched_commit_args, dbca_link);
-		dtx_free_dbca(dbca);
+	d_list_for_each_entry_safe(dbca, tmp, &dmi->dmi_dtx_batched_list,
+				   dbca_link) {
+		if (dbca->dbca_cont->sc_cos_shutdown)
```

Comment:
check 'sc_closing' and call dtx_free_dbca() only when sc_closing is true?

Reply:
The batched commit ULT may be called firstly, at that time sc_closing is not set.